### ISSUE TITLE
[apex] Mark Database DML statements in For Loop

### ISSUE THREAD

**Issue by JorneVL** [pmd/pmd#1713]
**Affects PMD Version:**
6.12

**Rule:**
[AvoidDmlStatementsInLoops](https://pmd.github.io/latest/pmd_rules_apex_performance.html#avoiddmlstatementsinloops)

**Description:**
The current check only warns for the insert word.
But this doesn't warn you when using the Database functions ( see sample 1 ).
This does the exact same thing.

Maybe we can add a rule that check for DML in a method name ( sample 2 ).
But this is something that is personal and can be a false notification.
If we add a rule then they can choose this for their own.

**Code Sample demonstrating the issue:**
Sample 1
```apex
public class Something {
    public void foo() {  
        for (Integer i = 0; i < 151; i++) {
            Account account;
            // ...
            Database.insert(account);
        }
    }
}

```

Sample 1
```apex
public class Something {
    public void foo() {  
        for (Integer i = 0; i < 151; i++) {
            Account account;
            // ...
            account.insertWithContact();
        }
    }
}

```

**Running PMD through:** *Other*
